src/components/apex-chart/apex-chart.ts

```diff
@@ -0,0 +1,81 @@
+import {bindable} from "aurelia-templating";
+import ApexCharts, {ApexOptions} from 'apexcharts';
+
+export interface ISelection {
+    series?:number;
+    dataPointIndex?:number;
+}
+
+export type ApexSeries = ApexAxisChartSeries | ApexNonAxisChartSeries
+
+export class ApexChart {
+    private _apex: HTMLDivElement;
+    private _myApexChart: ApexCharts;
+
+    @bindable options: ApexOptions;
+    @bindable series: ApexSeries;
+    @bindable selection: ISelection;
+    @bindable class:string;
+
+    private _attached:boolean = false;
+
+    attached() {
+        this._attached = true;
+        if (this.options) {
+            this._createChart();
+        }
+
+    }
+
+    detached() {
+        this._attached = false;
+        if (this._myApexChart) {
+            this._myApexChart.destroy();
+            this._myApexChart = null;
```

Comment:
`null` is not actually a valid type of `this._myApexChart`?

Reply:
It is null